include/common/enum_types.h

```diff
@@ -24,6 +24,15 @@ enum WasmEdge_ValType {
 #undef UseValType
 };
 
+union WasmEdge_ValTypeExt {
+  uint32_t TypeIdx;
```

Comment:
@apepkuss 

1. Actually, in this PR, the field is not used anywhere. This is just a temporary placeholder for the union in this PR, since the CI on Windows fails on a warning that the size of empty union type differs between C and C++. The final struct will be introduced in https://github.com/WasmEdge/WasmEdge/pull/2107. You may refer to the code changes in `enum.inc` and `enum_types.h` for that PR.

For you reference, the struct will be like
```
typedef struct WasmEdge_HeapType {
   enum WasmEdge_HeapTypeCode HeapTypeCode;
   uint32_t DefinedTypeIdx;
 } WasmEdge_HeapType;

 union WasmEdge_ValTypeExt {
   WasmEdge_HeapType HeapType;
 };

 typedef struct WasmEdge_FullValType {
   enum WasmEdge_ValTypeCode TypeCode;
   union WasmEdge_ValTypeExt Ext;
 } WasmEdge_FullValType;
```

The `HeapType` in `ValTypeExt` is only meaningful when the `TypeCode` is `Ref` or `RefNull`.


2. The union is defined as a C style code, so all the field types are `Copy` types. Besides, though the field types are `Copy`, dereferencing the value in the union is unsafe. Personally I will recommend defining the ValType in the Rust SDK as 
```
enum ValType {
    I32,
    I64,
    F32,
    F64,
    V128,
    RefNull(HeapType),
    Ref(HeapType),
}

enum HeapType {
    Func,
    Extern,
    Defined(u32),
}
```
, and then you can turned the struct defined in the generated rust binding code into this self defined enum by first checking the `ValTypeCode`, and then extract the specific fields in the union for different `ValTypeCode` so that the SDK user will not have to handle unsafe code.

3. For `NumType`, the `Ext` field will be meaningless, so if it is required, you can fill in any case as you like. For `RefType`, the `Ext` field will be meaningful, and you simply fill in HeapType in the union as it requires.

Reply:
Thanks for the detailed explanation. As for 3, I am still not sure what value of the `Ext` field is returned from the low-level runtime back to Rust SDK for a `NumType`. I'll check it further. Thanks!

Reply:
If a value is of `NumType`, it does not have any extension in the type definition, so there will not be `Ext` set for it, in other word, you don't have to interpret the `Ext` field if the value is of `NumType`. 

The generated struct for `FullValType` is `Sized`. Therefore, even if the `NumType` does not have `Ext`, because it shares the same struct definition with `RefType`, which has something set in the `Ext`, the `NumType` of `FullValType` still occupies some extra unused bytes of `Ext` fields in its memory layout. Since `NumType` does not use the `Ext` field, the bytes for the `Ext` fields can be any random value. And we don't have to pay attention to the `Ext` field for `NumType`, because it is not used. For `NumType`, you can still interpret the `Ext` field as `HeapType`, which is originally for `RefType`, and it can return a `HeapType` struct represented by those random bytes, and such interpretation is safe in rust, but just meaningless.